### PR TITLE
Feature/cu qthbph fang refactor

### PR DESCRIPTION
**Pull request by Yonimevicio**
- New player (fang have a visual issues)
- TestScene: Level1_TestNewFangModel_2.scene
- Meleegrunt and ranged updated to be able to be hitted by fang

### CODE REVIEW

**Comment by jcarlos0305** [comment]
Task linked: [CU-qthbph Fang refactor](https://app.clickup.com/t/qthbph)

File: Penteract/Assets/Scripts/AIMeleeGrunt.cpp [hunk 1]
```diff
@@ -308,46 +307,34 @@ void AIMeleeGrunt::OnAnimationFinished() {
 	}
 }
 
+void AIMeleeGrunt::ParticleHit(GameObject& collidedWith, void* particle, Player& player) {
+	if (!particle) return;
+	ComponentParticleSystem::Particle* p = (ComponentParticleSystem::Particle*)particle;
+	ComponentParticleSystem* pSystem = collidedWith.GetComponent<ComponentParticleSystem>();
+	if (pSystem) pSystem->KillParticle(p);
+	if (state == AIState::STUNNED && player.level2Upgrade) {
+		gruntCharacter.GetHit(99);
```
Comment (magalenyo) [inline Penteract/Assets/Scripts/AIMeleeGrunt.cpp:316]: Shouldn't this dmg be exposed? Or is it supposed to act as an instakill?

File: Penteract/Assets/Scripts/RangedAI.cpp [hunk 2]
```diff
@@ -214,51 +214,35 @@ void RangedAI::OnAnimationEvent(StateMachineEnum stateMachineEnum, const char* e
 		}
 	}
 }
+void RangedAI::ParticleHit(GameObject& collidedWith, void* particle, Player& player) {
+	if (!particle) return;
+	ComponentParticleSystem::Particle* p = (ComponentParticleSystem::Particle*)particle;
+	ComponentParticleSystem* pSystem = collidedWith.GetComponent<ComponentParticleSystem>();
+	if (pSystem) pSystem->KillParticle(p);
+	if (state == AIState::STUNNED && player.level2Upgrade) {
+		rangerGruntCharacter.GetHit(99);
```
Comment (magalenyo) [inline Penteract/Assets/Scripts/RangedAI.cpp:223]: Same as the comment above

**Review by magalenyo — DISMISSED** [review]
We're going to have to delete Onimaru_test.prefab and unnecessary scenes on merge.

**Review by jcarlos0305 — DISMISSED** [review]

**Review by FrancPS — DISMISSED** [review]

File: Penteract/Assets/Scripts/AIMeleeGrunt.cpp (hunk 1 above)
Comment (devildrake) [reply Penteract/Assets/Scripts/AIMeleeGrunt.cpp:322]: My guess is that it's instakill

**Review by devildrake — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Review by FrancPS — APPROVED** [review]

**Review by magalenyo — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Review by devildrake — APPROVED** [review]

**Pull request merged**